### ISSUE TITLE
[Electrs] Support for elements regtest in 0.18 is missing

### ISSUE THREAD

**Issue by tiero** [Blockstream/esplora#182]
Hi! Running Electrs for Liquid in regtest gives me `server failed: Error: failed to parse header`

I'm running the [latest master](https://github.com/Blockstream/electrs/commit/49cf7a7dffe8dccb3d4c5cb543aa05878abf086b) of electrs  in liquidregtest with this command

```sh
electrs -vvvv --network liquidregtest --daemon-dir /config --daemon-rpc-addr 10.10.0.11:18884 --cookie admin1:123 --http-addr 0.0.0.0:3002 --electrum-rpc-addr 0.0.0.0:60401 --cors *
```

and with this elements.conf with the latest [ release ](https://github.com/ElementsProject/elements/releases/tag/elements-0.18.1.5)

```
chain=liquidregtest
listen=1
txindex=0

[liquidregtest]
port=18886
rpcport=18884
rpcuser=admin1
rpcpassword=123
rpcallowip=0.0.0.0/0
rpcbind=0.0.0.0

mainchainrpcport=19001
mainchainrpchost=10.10.0.10
mainchainrpcuser=admin1
mainchainrpcpassword=123

validatepegin=0
initialfreecoins=2100000000000000

```

but electrs gives me this error

```
Config { log: StdErrLog { verbosity: Trace, quiet: false, timestamp: Off, modules: [], writer: "stderr", color_choice: Auto }, network_type: LiquidRegtest, db_path: "./db/liquidregtest", daemon_dir: "/config/liquidregtest", daemon_rpc_addr: V4(10.10.0.11:18884), cookie: Some("admin1:123"), electrum_rpc_addr: V4(0.0.0.0:60401), http_addr: V4(0.0.0.0:3002), monitoring_addr: V4(127.0.0.1:44224), jsonrpc_import: false, index_batch_size: 100, bulk_index_threads: 3, tx_cache_size: 10000, prevout_enabled: true, cors: Some("*"), precache_scripts: None, electrum_txs_limit: 100, parent_network: LiquidRegtest, parent_genesis_hash: "0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206", asset_db_path: None }
DEBUG - Server listening on 127.0.0.1:44224
DEBUG - Running accept thread
INFO - NetworkInfo { version: 180105, subversion: "/Elements Core:0.18.1.5/" }
INFO - BlockchainInfo { chain: "liquidregtest", blocks: 101, headers: 101, bestblockhash: "9a88e0471623ae68e0a6abfd5e539f4c4be65a8d837cf59b59065140aaedd876", pruned: false, verificationprogress: 1.0, initialblockdownload: Some(false) }
DEBUG - opening DB at "./db/liquidregtest/newindex/txstore"
DEBUG - 0 blocks were added
DEBUG - opening DB at "./db/liquidregtest/newindex/history"
DEBUG - 0 blocks were indexed
DEBUG - opening DB at "./db/liquidregtest/newindex/cache"
DEBUG - downloading all block headers up to 9a88e0471623ae68e0a6abfd5e539f4c4be65a8d837cf59b59065140aaedd876
TRACE - downloading 102 block headers
ERROR - server failed: Error: failed to parse header 000000a021cab1e5da4718ea140d9716931702422f0e6ad915c8d9b583cac2706b2a9000ac20a615d9b0d4df3e3ac2cb7018a07bd314d6bb715a57adead7c03e208b36587586595e01000000022200204ae81572f06e1b88fd5ced7a1a000945432e83e1551e6f721ee9c00b8cc332604a00000017a91472c44f957fc011d97e3406667dca5b1c930c4026870151014202fcba7ecf41bc7e1be4ee122d9d22e3333671eb0a3a87b5cdf099d59874e1940f02fcba7ecf41bc7e1be4ee122d9d22e3333671eb0a3a87b5cdf099d59874e1940f00010151
Caused by: parse failed: data not consumed entirely when explicitly deserializing
```

PS: If you want to replicate it's happening now on latest version of [Nigiri](https://github.com/vulpemventures/nigiri)

To replicate you should have docker & compose and do:

```
$ curl https://getnigiri.vulpem.com | bash
$ nigiri start --liquid
$ docker logs resources_electrs-liquid_1
```

**Comment by tiero** [comment]
PPS: apparently new flag `--parent-network` has been added, giving that I have the same result, even if now connect to Regtest instead of Bitcoin and returns the right `parent_genesis_hash`

```

Config { log: StdErrLog { verbosity: Trace, quiet: false, timestamp: Off, modules: [], writer: "stderr", color_choice: Auto }, network_type: LiquidRegtest, db_path: "./db/liquidregtest", daemon_dir: "/config/liquidregtest", daemon_rpc_addr: V4(10.10.0.11:18884), cookie: Some("admin1:123"), electrum_rpc_addr: V4(0.0.0.0:60401), http_addr: V4(0.0.0.0:3002), monitoring_addr: V4(127.0.0.1:44224), jsonrpc_import: false, index_batch_size: 100, bulk_index_threads: 3, tx_cache_size: 10000, prevout_enabled: true, cors: Some("*"), precache_scripts: None, electrum_txs_limit: 100, parent_network: Regtest, parent_genesis_hash: "0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206", asset_db_path: None }
DEBUG - Server listening on 127.0.0.1:44224
DEBUG - Running accept thread
WARN - reconnecting to bitcoind: {"code":-28,"message":"Loading block index..."}
WARN - reconnecting to bitcoind: {"code":-28,"message":"Loading wallet..."}
WARN - reconnecting to bitcoind: {"code":-28,"message":"Loading wallet..."}
INFO - NetworkInfo { version: 180105, subversion: "/Elements Core:0.18.1.5/" }
INFO - BlockchainInfo { chain: "liquidregtest", blocks: 0, headers: 0, bestblockhash: "00902a6b70c2ca83b5d9c815d96a0e2f4202179316970d14ea1847dae5b1ca21", pruned: false, verificationprogress: 0.0, initialblockdownload: Some(true) }
WARN - waiting for bitcoind sync to finish: 0/0 blocks, vertification progress: 0.000%
DEBUG - opening DB at "./db/liquidregtest/newindex/txstore"
DEBUG - 0 blocks were added
DEBUG - opening DB at "./db/liquidregtest/newindex/history"
DEBUG - 0 blocks were indexed
DEBUG - opening DB at "./db/liquidregtest/newindex/cache"
DEBUG - downloading all block headers up to 29995d9c1f00a2c0d3f258cf6dc678adb269cb5f21716370d764292b0452045a
TRACE - downloading 102 block headers
ERROR - server failed: Error: failed to parse header 000000a021cab1e5da4718ea140d9716931702422f0e6ad915c8d9b583cac2706b2a9000ac20a615d9b0d4df3e3ac2cb7018a07bd314d6bb715a57adead7c03e208b3658d9c35a5e01000000022200204ae81572f06e1b88fd5ced7a1a000945432e83e1551e6f721ee9c00b8cc332604a00000017a91472c44f957fc011d97e3406667dca5b1c930c4026870151014202fcba7ecf41bc7e1be4ee122d9d22e3333671eb0a3a87b5cdf099d59874e1940f02fcba7ecf41bc7e1be4ee122d9d22e3333671eb0a3a87b5cdf099d59874e1940f00010151
Caused by: parse failed: data not consumed entirely when explicitly deserializing
```

**Comment by tiero** [comment]
Ok, I guess I found the reason of the error. The correct chain to use on Elements side is `elementsregtest`, but doing so it will break electrs because apparently doesn't support the new chain yet as per here `https://github.com/Blockstream/electrs/blob/new-index/src/chain.rs#L76`

**Comment by stevenroose** [comment]
@tiero as explained [on the elements readme](https://github.com/ElementsProject/elements/#modes), "elementsregtest" is not really a special name. Any name that is not in that list of special networks names can be chosen. So it's quite hard for tools like espora to support these kind of chains because they can have any name.

If for you it doesn't matter, you can run elements by using `-chain=liquidregtest` and it will probably be compatible with esplora.

**Comment by tiero** [comment]
I'm talking about electrs, not esplora the frontend. I posted here because the issues [here](https://github.com/Blockstream/electrs) are deactivated.

Btw Yes you are right. Elements running with `liquidregtest` it's totally fine.

The problem is in `electrs` that gives me the error in the first/second comment. Any clue?

**Comment by tiero** [comment]
FYI Rolling back to 0.17.0.3 solved the issue temporarily for my use case. I'll keep an eye here to see if it gets solved so I can upgrade to 0.18 again. Thanks

**Comment by greenaddress** [comment]
@tiero we added support for bitcoin regtest and liquid regtest to esplora in master

See instructions on the README.md

Will close this issue, feel free to reopen or open a new one if you find any issue or have feedback.

thanks!
